src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx

```diff
@@ -3,170 +3,15 @@
-
-        setIsLanguageReferenceOpen(true);
-        document.body.appendChild(languageReferenceContainer);
-
-        const element = (
-          <EuiWrappingPopover
-            id="languageReferencePopover"
-            button={anchorElement}
-            isOpen={true}
-            closePopover={onCloseLanguageReference}
-            panelPaddingSize="s"
-            anchorPosition="downLeft"
-            anchorClassName="euiFormControlLayout__append"
-          >
-            <EuiPopoverTitle>
-              <FormattedMessage
-                id="data.query.queryBar.syntaxOptionsTitle"
-                defaultMessage="Syntax options"
-              />
-            </EuiPopoverTitle>
-            <div style={{ width: '350px' }}>
-              <EuiText size="s">
-                <p>
-                  <FormattedMessage
-                    id="data.query.queryBar.syntaxOptionsDescription"
-                    defaultMessage="The {docsLink} (DQL) offers a simplified query
-              syntax and support for scripted fields."
-                    values={{
-                      docsLink: (
-                        <EuiLink href={osdDQLDocs} target="_blank">
-                          {dqlFullName}
-                        </EuiLink>
-                      ),
-                    }}
-                  />
-                </p>
-              </EuiText>
-            </div>
-          </EuiWrappingPopover>
-        );
-
-        ReactDOM.render(element, languageReferenceContainer);
-      }
-    },
-    iconType: 'iInCircle',
-  };
-
-  const languageToggle: QueryControl = {
-    id: 'languageToggle',
-    label: i18n.translate('discover.queryControls.languageToggle', {
-      defaultMessage: 'Toggle',
-    }),
-    testId: 'languageToggle',
-    ariaLabel: i18n.translate('discover.queryControls.languageToggle', {
-      defaultMessage: `Language Toggle`,
-    }),
-    run: () => {
-      setIsCollapsed(!isCollapsed);
-      props.onToggleCollapse();
-    },
-    iconType: isCollapsed ? 'expand' : 'minimize',
-  };
-
-  const queryControls =
-    props.queryLanguage === 'PPL' || props.queryLanguage === 'SQL'
-      ? [languageReference, languageToggle]
-      : [languageReference];
-
-  if (props.additionalControls) {
-    queryControls.push(...props.additionalControls);
-  }
+import React from 'react';
+import { EuiFlexItem } from '@elastic/eui';
 
+export const QueryControls = (props: { queryControls: React.ReactElement[] }) => {
   return (
-    <EuiFlexGroup responsive={false} gutterSize="s" alignItems="center">
-      {queryControls.map((queryControl) => (
-        <EuiFlexItem grow={false}>
-          <EuiButtonIcon
-            iconType={queryControl.iconType}
-            aria-label={queryControl.ariaLabel}
-            onClick={(event) => queryControl.run(event.currentTarget)}
-          />
-        </EuiFlexItem>
+    <>
+      {props.queryControls.map((queryControl) => (
+        <EuiFlexItem grow={false}>{queryControl}</EuiFlexItem>
```

Comment:
i think it's better to add `keys` attribute

Reply:
added